### DISCUSSION TITLE
Want to encrypt password passed in "sql_alchemy_conn" in airflow.cfg

### DISCUSSION THREAD

**Discussion by OmkarShinde15** [apache/airflow#36708] [Q&A]
I have "sql_alchemy_conn = mysql://airflow_user:Airflow123@x.x.x.x:3306/airflow_db_one" variable inside airflow.cfg which is used to connect backend database.

I want to encrypt/hash my password "Airflow123" from airflow.cfg file

I tried fernet, I placed "fernet_key", "encrypted_mysql_pass" and "sql_alchemy_conn = mysql://airflow_user:{encrypted_mysql_pass}@x.x.x.x:3306/airflow_db_one" in airflow.cfg file.

with above change i got error stating "Access denied"

**Comment by potiuk** [comment]
You should use secrets backends for that: https://airflow.apache.org/docs/apache-airflow/stable/security/secrets/secrets-backend/index.html There are various implementations available and you can roll your own.

You can also use _CMD version of env variable foro airflow connection to retrieve your credentials without storing them in local config file, and use for example workload identity (look it up for various cloud implementation) to make sure only your workloads can retrieve that information: https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html
